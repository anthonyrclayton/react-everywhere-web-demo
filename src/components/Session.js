import React, { Component } from 'react';

class Session extends Component {
  render() {
    return(
      <div className="session">
        <h2>{this.props.title}</h2>
        <div>{this.props.description}</div>
      </div>
    )
  }
}

Session.propTypes = {
  title: React.PropTypes.string.isRequired,
  description: React.PropTypes.string.isRequired
}

export default Session;
