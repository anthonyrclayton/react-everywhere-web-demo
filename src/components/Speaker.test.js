import React from 'react';
import Session from './Session';
import renderer from 'react-test-renderer';

const session = {
    "Id": 5445,
    "SessionTime": "0001-01-01T00:00:00",
    "SessionStartTime": "2017-01-10T08:00:00",
    "SessionEndTime": "2017-01-10T12:00:00",
    "Room": null,
    "Rooms": [
      "Indigo Bay"
    ],
    "Title": "7 Languages in 7 Hours",
    "Abstract": "Great Session",
    "SessionType": "Pre-Compiler",
    "Tags": [
      "Ruby/Rails",
      "Testing",
      "Functional Programming",
      "Other"
    ],
    "Category": "Programming Principles",
    "Speakers": [
      {
        "Id": "df1a9b69-1b7a-4224-bf28-2796a6cbed61",
        "FirstName": "Amber",
        "LastName": "Conville",
        "GravatarUrl": "//www.gravatar.com/avatar/7c2a83b93fe0e5f2ed4c95c59367259c"
      }
    ]
}

it('renders without crashing', () => {
  const output = renderer.create(<Session {...session}/>);

  expect(output).toMatchSnapshot()
});
