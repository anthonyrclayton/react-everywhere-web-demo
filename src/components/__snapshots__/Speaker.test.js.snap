exports[`test renders without crashing 1`] = `
<div
  className="session"
  onClick={[Function]}>
  <h2>
    7 Languages in 7 Hours
  </h2>
  <div
    className="Session-details">
    <div>
      <div
        className="speaker">
        <div>
          <img
            alt="Amber"
            src="//www.gravatar.com/avatar/7c2a83b93fe0e5f2ed4c95c59367259c" />
        </div>
        <h9>
          Amber
          Â 
          Conville
        </h9>
      </div>
    </div>
    <div
      className="Session-abstract">
      Great Session
    </div>
  </div>
</div>
`;
